File: frontend/src/types/game.ts
export interface Console {
  id: string;
  name: string;
  manufacturer?: string;
  release_year?: number;
  icon_url?: string;
  emulator_core: string;
  supported_extensions: string[];
}

export interface Game {
  id: string;
  console_id: string;
  title: string;
  rom_filename: string;
  rom_size_bytes: number;
  cover_url?: string;
  description?: string;
  release_year?: number;
  developer?: string;
  genre?: string;
  created_at: string;
  uploaded_by?: string;
}

export interface GameListResponse {
  games: Game[];
  total: number;
}

export interface UserGameStats {
  id: string;
  user_id: string;
  game_id: string;
  total_playtime_seconds: number;
  last_played?: string;
  first_played: string;
  play_count: number;
}

File: backend/src/models/game.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Console {
    pub id: String,
    pub name: String,
    pub manufacturer: Option<String>,
    pub release_year: Option<i32>,
    pub icon_url: Option<String>,
    pub emulator_core: String,
    pub supported_extensions: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Game {
    pub id: Uuid,
    pub console_id: String,
    pub title: String,
    pub rom_filename: String,
    pub rom_size_bytes: i64,
    pub cover_url: Option<String>,
    pub description: Option<String>,
    pub release_year: Option<i32>,
    pub developer: Option<String>,
    pub genre: Option<String>,
    pub created_at: DateTime<Utc>,
    pub uploaded_by: Option<Uuid>,
}

#[derive(Debug, Deserialize)]
pub struct GameUploadRequest {
    pub console_id: String,
    pub title: String,
    pub description: Option<String>,
    pub release_year: Option<i32>,
    pub developer: Option<String>,
    pub genre: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct GameListResponse {
    pub games: Vec<Game>,
    pub total: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct UserGameStats {
    pub id: Uuid,
    pub user_id: Uuid,
    pub game_id: Uuid,
    pub total_playtime_seconds: i64,
    pub last_played: Option<DateTime<Utc>>,
    pub first_played: DateTime<Utc>,
    pub play_count: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct PlaySession {
    pub id: Uuid,
    pub user_id: Uuid,
    pub game_id: Uuid,
    pub started_at: DateTime<Utc>,
    pub ended_at: Option<DateTime<Utc>>,
    pub duration_seconds: Option<i32>,
}

#[derive(Debug, Deserialize)]
pub struct TrackSessionRequest {
    pub game_id: Uuid,
    pub session_duration: i32,
}
